lib/schema.js

```diff
@@ -187,16 +188,39 @@ export default class Schema {
    *
    */
   index(prop, options = {}){
-    if (!this.descriptor[prop]){
-      throw new Error('Descriptor not found for this field');
-    }
+    if (Array.isArray(prop)){
+      _.each(prop, p => {
+        if (!this.descriptor[p]){
+          throw new Error(`Descriptor not found for ${p}`);
+        }
+      });
+
+      let indexType = options.indexType ? options.indexType: 'single';
+      let indexName = options.indexName;
+      if (!indexName){
+        const reg = new RegExp('^\\d+$');
+        indexName = _.map(prop, p => {
+          if (reg.test(p)) {
+            name = p[p.length - 2];
+          }
+          return inflection.singularize(p);
+        });
+        indexName = indexName.join('_and_');
+      }
 
-    this.descriptor[prop].index = true;
-    this.descriptor[prop] = _.extend(this.descriptor[prop], options);
+      this.indexes[indexName] = {path: prop, name: indexName, indexType: indexType, compound: true};
+    } else {
+      if (!this.descriptor[prop]){
+        throw new Error('Descriptor not found for this field');
+      }
 
-    let fullDes = {};
-    fullDes[prop] = this.descriptor[prop];
-    this._getIndexes(fullDes);
+      this.descriptor[prop].index = true;
+      this.descriptor[prop] = _.extend(this.descriptor[prop], options);
```

Comment:
`_.extend` mutates the object so I'd do `_.extend({}, this.descriptor[prop], options);`